packages/lodestar-validator/src/util/misc.ts

```diff
@@ -1,3 +1,11 @@
 export function sleep(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
+
+export function abortableTimeout(
```

Comment:
An abortableTimeout should start the timeout for you and cancel it on abort
```ts
export function abortableTimeout(callback, timeoutMs, signal): Timeout {
  const timeout = setTimeout(callback, timeoutMs)
  signal?.addEventListener("abort", () => {
    clearTimeout(timeout)
  }, {once: true});

  return timeout;
}
```
Or something like this I think would make more sense

Reply:
You could even use this abortable sleep function
https://github.com/ChainSafe/lodestar/blob/4295ce0f4c3f2e8e8a39e6ce01f813219b668abc/packages/lodestar/src/util/sleep.ts#L8-L30

And use the sleep as if it was a setTimeout do 
```ts
sleep(1000, signal).then(
  () => {
    // timeout completed, do stuff
  },
  () => {
    // timeout aborted, do stuff
  }
)
```
I would like to know what others think about this approach

Reply:
Not sure how everyone else feels, but I like the first option you brought up in your previous comment more because it's simpler.  If we were to go that route though, we would also need an abortCallback or similar to handle logging, closing the EventSource, etc.

Also, `sleep` would force the whole timeout to be a Promise, which I'm not sure would be applicable here:
https://github.com/ChainSafe/lodestar/pull/1440/files#diff-cd304f421d116e19fd26363114c73e94R179

Reply:
Or would it be applicable there?  I might not be seeing it correctly.